### DISCUSSION TITLE
How can I get charging time in CCCV charging?

### DISCUSSION THREAD

**Discussion by YouHyin** [pybamm-team/PyBaMM#2312] [Q&A]
I really need a method to obtain a charging time from pybamm to implement Bayesian optimization that minimizes charging time. It means the total time spent charging.
![2325](https://user-images.githubusercontent.com/113442395/191674762-95de828a-39d1-400b-ba35-22b4f3c50afb.png)

Is there a code that I can find the charging time as value other than finding the charging time with the intercept of the x-coordinate on the graph?

**Comment by valentinsulzer** [comment]
How are you defining charging time?

**Reply by YouHyin** [reply]
First of all, thank you very much for your answer. It means the time from the beginning to the completion of charging in cccv charging. That means 1450.14 seconds in this picture. My purpose is to create a function that receives this 1450.14 seconds as a return value. How do I get this by code rather than by x-intercept of plot? I really need your help.

**Reply by rtimms** [reply]
If that's your whole solution you can just do `solution["Time [s]"].entries[-1]` to get the final time.

**Reply by YouHyin** [reply]
Thank you so much It was very helpful.
